run.go

```diff
@@ -139,38 +135,102 @@ var commandInspect = cli.Command{
 	Action: actionInspect,
 }
 
-func guessType(filename string) string {
-	switch {
-	case strings.Contains(filename, "://"):
-		return TypeURL
-	case strings.HasSuffix(filename, ".js"):
-		return TypeJS
-	default:
-		return ""
+func looksLikeURL(str []byte) bool {
+	// Try to parse string as URL
+	u, err := url.Parse(strings.TrimSpace(string(str)))
```

Comment:
It's mostly that url.Parse() does a lot more than check if it looks like a URL - there's a couple of allocations of data we're immediately throwing away, and an invalid URL here is instead getting treated as a JS script and will yield a JS syntax error, rather than getting passed to the URL runner where a parse error would be properly logged as such.